File: Daily Test/Java/String And Character Matrix 12.4.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int R = sc.nextInt(), C = sc.nextInt();
        String[][] matrix = new String[R][C];

        for (int i = 0; i < R; i++)
            for (int j = 0; j < C; j++)
                matrix[i][j] = sc.next();

        int X = sc.nextInt(), Y = sc.nextInt();
        String target = matrix[X - 1][Y - 1];
        int n = (int)Math.sqrt(target.length());

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                System.out.print(target.charAt(i * n + j) + " ");
            }
            System.out.println();
        }
    }
}

File: Daily Test/C++/String And Character Matrix 12.4.2025.cpp
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;

int main() {
    int R, C;
    cin >> R >> C;
    vector<vector<string>> matrix(R, vector<string>(C));

    for (int i = 0; i < R; ++i)
        for (int j = 0; j < C; ++j)
            cin >> matrix[i][j];

    int X, Y;
    cin >> X >> Y;

    string target = matrix[X - 1][Y - 1];
    int n = sqrt(target.size());

    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < n; ++j)
            cout << target[i * n + j] << " ";
        cout << endl;
    }

    return 0;
}